### PR TITLE
Modify Nc4Iosp to allow reading of CDF-5 files

### PR DESCRIPTION
**Pull request by DennisHeimbigner**
Modify Nc4Iosp to allow reading of CDF-5 files
by treating them as netcdf-4 files.

### CODE REVIEW

**Comment by cwardgar** [comment]
Thus far, we've avoided placing any I/O code in `NetcdfFile`, preferring instead to put that code in the IOSPs. Is there a reason that the additions can't be limited to `Nc4Iosp` and `Nc4prototypes`?

**Comment by cwardgar** [comment]
Also, a unit test for CDF-5 would be great, assuming that we can find a small (< 1MB) test file.

**Comment by DennisHeimbigner** [comment]
Modified to:
1. Create new file ./cdm/src/main/java/ucar/nc2/iosp/NCheader.java
   that becomes super class for H4header, H5header, N3header.
2. Move checkFileType from NetcdfFile to NCheader
3 Modify  H4header, H5header, N3header so that isValidFile()
   used NCheader/checkFileType()

**Comment by DennisHeimbigner** [comment]
Apparently my travis problems were transitory. So once
the branch update checks finish (assuming success)
this pr is ready to merge.

File: cdm/src/main/java/ucar/nc2/NetcdfFile.java [hunk 1]
```diff
@@ -129,6 +102,26 @@
...
+  static public final int  NC_FORMAT_64BIT_DATA = (5);
+
+  // Extensions
+  static public final int  NC_FORMAT_HDF4 = (0x7005);
+
+  /* Aliases */
+  static public final int  NC_FORMAT_64BIT = (NC_FORMAT_64BIT_OFFSET);
+  static public final int  NC_FORMAT_CDF5 = (NC_FORMAT_64BIT_DATA);
+  static public final int  NC_FORMAT_HDF5 = (NC_FORMAT_NETCDF4);
+
```
Comment (cwardgar) [inline cdm/src/main/java/ucar/nc2/NetcdfFile.java]: This is no longer necessary, now that we have `NCheader`.

File: cdm/src/main/java/ucar/nc2/NetcdfFile.java [hunk 2]
```diff
@@ -2546,24 +2539,59 @@ static private void appendStructureName(StringBuilder sbuff, CDMNode n, String r
...
+	    return 0; // unknown
+      if(magic[0] == (byte)0x89
+       && magic[1] == (byte)'H'
+       && magic[2] == (byte)'D'
+       && magic[3] == (byte)'F') {
+	    return NC_FORMAT_HDF5;
+      }
+      filePos = (filePos == 0) ? 512 : 2 * filePos;
+    }
+    return 0;
```
Comment (cwardgar) [inline cdm/src/main/java/ucar/nc2/NetcdfFile.java]: Nor is this.

File: netcdf4/src/main/java/ucar/nc2/jni/netcdf/Nc4prototypes.java [hunk 3]
```diff
@@ -104,9 +104,12 @@ Use this in mode flags for both nc_create() and nc_open(). */
  *  by nc_inq_format. 
  */
   int NC_FORMAT_CLASSIC = (1);
-  int NC_FORMAT_64BIT = (2);
+  int NC_FORMAT_64BIT_OFFSET = (2);
+  int NC_FORMAT_64BIT = (NC_FORMAT_64BIT_OFFSET); // Alias
   int NC_FORMAT_NETCDF4 = (3);
   int NC_FORMAT_NETCDF4_CLASSIC = (4);
+  int NC_FORMAT_64BIT_DATA = (5);
+  int NC_FORMAT_CDF5 = (NC_FORMAT_64BIT_DATA); // alias
```
Comment (cwardgar) [inline netcdf4/src/main/java/ucar/nc2/jni/netcdf/Nc4prototypes.java]: We may not need these changes either. Or, at the least, these variables should be assigned the values from `NCheader`. Or possibly vice versa. Basically, I'm complaining about duplication of constant values.

**Pull request merged**